src/simulations/modeling/adapter.py

```diff
@@ -588,3 +593,31 @@ def bounds(measurement, uncertainty):
         warnings.append(warning)
         logger.warning(warning)
     return operations, warnings, errors
+
+
+def get_ec_exchange_reaction(exchange_reactions, consumption):
+    """
+        Finds the corresponding exchange reaction in the ecModel to modify.
+
+        Parameters
+        ----------
+        exchange_reactions: set(cobra.Reaction)
+            ecModels by construction have 2 exchange reactions: one for consumption
+            (with formula --> X, i.e. only 1 product) and one for production (with
+            formula X -->, i.e. only 1 reactant).
+        consumption: bool
+            True if the consumption exchange reaction is needed and False if the
+            production exchange reaction is needed instead.
+
+        Returns
+        -------
+        ec_exchange_reaction: cobra.Reaction
+            The desired exchange reaction.
+        """
+    ec_exchange_reaction = []
+    for reaction in exchange_reactions:
+        if (reaction.products and consumption) or (
+            reaction.reactants and not consumption
+        ):
+            ec_exchange_reaction.append(reaction)
+    return ec_exchange_reaction
```

Comment:
It seems strange that this returns a list, but I see how it fits in with the existing error handling. I guess it depends on it getting valid input.

I think we should encapsulate the whole "get me the exchange reaction" logic now, and move it to another module. `cobra_helpers.py` seems a good candidate. It could accept `(model, metabolite, is_ec_model, consumption)` (consumption only being relevant when `is_ec_model` is True) and always return a single reaction. We can then abstract away the current error handling by raising some exception (I guess it's rare that a metabolite incorrectly has two exchange reactions in non-ecModels).

Reply:
Agree with your idea! But I would say let's address it in a separate PR, as the scope is broader than for just ecModels.